File: src/main/java/pt/haslab/dql/tictactoe/agents/HumanAgent.java
package pt.haslab.dql.tictactoe.agents;

import pt.haslab.dql.tictactoe.game.Board;

import java.util.Scanner;

/**
 * Implements an agent that receives the next move from
 * the console, thus enabling human players.
 * Created by nunomachado on 06/08/17.
 */
public class HumanAgent extends Agent {
    private static Scanner in;

    public HumanAgent(){
        in = new Scanner(System.in);
    }

    //Read next move from the console (i.e. from human input)
    public String act(Board state) {
        int row = in.nextInt();
        int col = in.nextInt();
        String action = row + " " + col;

        return action;
    }

    @Override
    public String toString()
    {
        return AgentType.HUMAN.toString();
    }
}

File: tictactoe/agents/human_agent.py
from tictactoe.agents.agent import Agent
from tictactoe.game.board import Board


class HumanAgent(Agent):

    def __init__(self):
        self.inp = input

    def make_move(self, state: Board) -> str:
        row = int(self.inp("Enter row: "))
        col = int(self.inp("Enter column: "))
        return f"{row} {col}"

    def __str__(self):
        return "HUMAN"
